rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/implementation/DropBoxHelper.java

```diff
@@ -0,0 +1,234 @@
+package com.stnetix.ariaddna.externalcloudapi.implementation;
+
+import com.dropbox.core.DbxAppInfo;
+import com.dropbox.core.DbxException;
+import com.dropbox.core.DbxRequestConfig;
+import com.dropbox.core.v2.DbxClientV2;
+import com.dropbox.core.v2.files.*;
+import com.dropbox.core.v2.users.SpaceUsage;
+import com.google.gson.JsonObject;
+import com.sun.jndi.toolkit.url.Uri;
+
+import javax.swing.*;
+import java.awt.*;
+import java.io.*;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Locale;
+
+/**
+ * Created by LugovoyAV on 06.09.2017.
+ */
+//запрос на аторизацию
+//https://www.dropbox.com/1/oauth2/authorize?response_type=code&client_id=ioiju89iyfxjy4h
+//TODO непонятно пока что возвращать JsonObject
+public class DropBoxHelper {
+    final static String APP_KEY = "ioiju89iyfxjy4h";
+    final static String APP_SECRET = "uys63lipze6ig4v";
+    DbxAppInfo appInfo;
+    String accessToken;
+    DbxClientV2 client;
+
+    public String getAccessToken() {
+        return accessToken;
+    }
+
+    public void setAccessToken(String accessToken) {
+        this.accessToken = accessToken;
+        DbxRequestConfig config = DbxRequestConfig.newBuilder("LexsusApp")
+                .withUserLocale(Locale.getDefault().toString())
+                .build();
+        client = new DbxClientV2(config, accessToken);
+        try {
+            System.out.println("Linked account: " + client.users().getCurrentAccount().getName());
+            System.out.println("email: " + client.users().getCurrentAccount().getEmail());
+        } catch (DbxException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public DropBoxHelper(String accessToken) {
+        this.accessToken = accessToken;
+
+        appInfo = new DbxAppInfo(APP_KEY, APP_SECRET);
+
+        DbxRequestConfig config = DbxRequestConfig.newBuilder("LexsusApp")
+                .withUserLocale(Locale.getDefault().toString())
+                .build();
+        client = new DbxClientV2(config, accessToken);
+        try {
+            System.out.println("Linked account: " + client.users().getCurrentAccount().getName());
+            System.out.println("email: " + client.users().getCurrentAccount().getEmail());
+        } catch (DbxException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public DropBoxHelper() {
+        this.accessToken = "";
+        appInfo = new DbxAppInfo(APP_KEY, APP_SECRET);
+
+    }
+
+    //TODO что такое path - файл на облаке или файл на локальном hdd
+    public JsonObject downloadFile(String path)
+    {
+        try
+        {
+            FileOutputStream downloadFile = new FileOutputStream(path);
+            try {
+                FileMetadata metadata = client.files().downloadBuilder("/"+path).download(downloadFile);
```

Comment:
This is don't FileMetadata on VUFS, this is chunk. Metadata (in VUFS) it's data in system DB about files properties.
Chunk it is part of some real file on client device. File Splitter Service make chunk and transmit it to Cloud Loader Service. Cloud Loader Service use some API adapter (for example Dropbox) make root directory for ariADDna storage on the cloud. And after it load chunks to the cloud.